Snapshot terminology update (#850)
chromaui/chromatic-docs: src/content/snapshot/snapshots.md

# Snapshots

> A **snapshot** is the unit of work executed by Chromatic to execute a [UI Test](/docs#test-how-uis-look--function). This work can be taking a screenshot, capturing accessibility data, copying such information from one test to another, or even bypassing such work completely if it is found to be unnecessary.

Reviewer: ```suggestion
A **snapshot** is the unit of work executed by Chromatic to run a [UI Test](/docs#test-how-uis-look--function). This work can be taking a screenshot, capturing accessibility data, copying such information from one test to another, or even bypassing such work completely if it is found to be unnecessary.
```

<div class="aside">

Looking for information on snapshot billing? [Go to billing docs](/docs/billing)

</div>

The type of work performed by Chromatic is determined by the following factors:

- **Visual vs. Accessibility:** A [visual test](/docs/visual) captures the rendered state of the UI, whereas an [accessibility test](/docs/accessibility) captures accessibility information about the UI. Both types of tests take a snapshot and compare it to a baseline snapshot to detect changes.

> - **Capture vs. Turbo:** A captured snapshot performs the full suite of work, including taking screenshots or capturing accessibility data, and running diffs. A [turbo snapshot](/docs/turbosnap) skips unnecessary work if it is determined that the associated code has not changed, speeding up the testing process.

Reviewer: ```suggestion
- **Capture vs. TurboSnap:** A captured snapshot performs the full suite of work, including taking screenshots or capturing accessibility data, and running diffs. A [TurboSnap](/docs/turbosnap) snapshot avoids unnecessary work if it is determined that the associated code has not changed, speeding up the testing process.
```

> These dimensions stack to create a snapshot. For example, you can have a "captured visual snapshot" or a "turbo accessibility snapshot". The short forms are "visual snapshot" and "accessibility turbosnap", respectively.

Reviewer: You can't have a "turbo accessibility snapshot" because we do not double dip on billing. If you have ART and VRT enabled and TurboSnap decides the test doesn't need to run, we'll only charge you for 1 turbosnap not two.

Author: ah, good point

## Snapshot-compatible tests

A **test** represents a piece of UI rendered in a specific state that Chromatic can capture as a snapshot. Chromatic supports the following types of tests:

> - **Storybook Stories:** By default, a story captures the rendered state of a UI component. Therefore Chromatic uses stories as is for UI testing.

Reviewer: ```suggestion
- **Storybook Stories:** By default, a story captures the rendered state of a UI component. Chromatic uses stories as is for UI testing.
```

> - **Playwright & Cypress End-to-End (E2E) tests:** Chromatic [archives](/docs/faq/what-is-archive) these E2E tests during Playwright/Cypress test runs, and then replays them in the cloud to run visual tests.

Reviewer: We archive **all tests** including for stories. It feels weird to talk about this implementation detail only for e2e and Vitest when the same thing happens for stories. 

Is archive an implementation detail or a critical feature? Probably the latter.

Author: yea, fair. I'll drop it

- **Vitest browser mode tests:** Chromatic [archives](/docs/faq/what-is-archive) these tests during Vitest test runs, and then replays them in the cloud to run visual tests.

Each test can generate multiple snapshots, varying by [browser](/docs/browsers#browser-support) (or mobile simulators and emulators), [theme](/docs/themes), [viewport size](/docs/viewports), and other configurations you define.

With [Playwright](/docs/playwright/targeted-snapshots), [Cypress](/docs/cypress/targeted-snapshots), and [Vitest](/docs/vitest/targeted-snapshots), you can even capture multiple visual snapshots within a single test for even more granular analysis.

## How are snapshots captured?

Chromatic captures snapshots by following these steps:

<details>
<summary>1. Load each test in the designated device and viewport</summary>

Chromatic's Capture Cloud leverages a fleet of standardized [browsers](/docs/browsers) and mobile emulators to load all of your tests (either stories or archives) in parallel, at the specified [viewport size](/docs/viewports).

</details>

<details>
<summary>2. Wait for the test to render</summary>

> Capture Cloud uses underlying browser APIs combined with our own set of heuristics to determine when the UI has "loaded".

Reviewer: ```suggestion
Capture Cloud uses underlying browser APIs combined with our own set of [heuristics](/features/steadysnap) to determine when the UI has "loaded" and is ready to be captured.
```

![snapshot with interactions and delay](../../images/snapshot-with-interactions-and-delay.png)

### View visual snapshot for a test

The component screen allows you to switch between 'Canvas' and 'Snapshot'. Under the Snapshot tab, you'll find the image captured by Chromatic's cloud browser. This is exactly what the browser _saw_ when it rendered the test. This could be the component's _initial_ or _intermediate_ state.

![Component screen with snapshot](../../images/component-snapshot.png)

<div class="aside">Tip: Click the arrow icon in the top right to open the test. Stories open in your published Storybook.</div>

### View live rendered test using the Canvas

**Canvas** is an interactive environment that renders your test code live in the browser. These are fully inspectable, enabling you to troubleshoot changes and errors effectively.

> With Storybook, it displays the story, and with Playwright, Cypress and Vitest, it shows the [archive](/docs/faq/what-is-archive) of your test. It visualizes the _final state_ of the test UI.

Reviewer: ```suggestion
With Storybook, it displays the story. With Playwright, Cypress and Vitest, it shows the [archive](/docs/faq/what-is-archive) of your test. In all cases, you see the _final state_ of the test UI.
```

![Component screen in Canvas mode](../../images/component-canvas.png)

### Differences in visual snapshot vs Canvas

> The visual snapshot might differ from the `Canvas` due to various reasons, such as JavaScript execution being blocked during capture, which can prevent certain elements from being captured, or the use of the `isChromatic()` function to alter rendering on Chromatic.

Reviewer: ```suggestion
The visual snapshot might differ from the `Canvas` for various reasons, such as JavaScript execution being blocked during capture, which can prevent certain elements from being captured. Or the use of the `isChromatic()` function to alter rendering on Chromatic.
```
